MaintenanceManager/MaintenanceManager.cpp

```diff
@@ -216,67 +217,234 @@ namespace WPEFramework {
          }
 
         void MaintenanceManager::task_execution_thread(){
-            int task_count=3;
-            int i=0;
+            uint8_t i=0;
             string cmd="";
+            bool internetConnectStatus=false;
 
-            LOGINFO("Reboot_Pending :%s",g_is_reboot_pending.c_str());
+            /* cleanup if not empty */
+            if(!tasks.empty()){
+                tasks.erase (tasks.begin(),tasks.end());
+            }
 
-            bool internetConnectStatus = isDeviceOnline();
+            /* Controlled by CFLAGS */
+#if defined(SUPPRESS_MAINTENANCE)
+            bool activationStatus=false;
+            bool skipFirmwareCheck=false;
 
-            MaintenanceManager::_instance->onMaintenanceStatusChange(MAINTENANCE_STARTED);
-            /*  In an unsolicited maintenance we make sure only after
-             *  after DCM task activities are started.
-             */
+            /* Activation check */
+            activationStatus = isDeviceActivated(skipFirmwareCheck);
```

Comment:
Pls change isDeviceActivated to getActivationStatus to better reflect the purpose.